cli/src/main.rs

```diff
@@ -5,45 +5,45 @@ use std::{
     io::{Read, Write},
 };
 
-use clap::Parser;
 use color_eyre::{owo_colors::OwoColorize, Result};
 
-mod args;
 mod cargo;
 mod metadata;
 mod path;
+mod config;
 
-use args::Args;
 use metadata::{Metadata, FILE_NAME};
+use crate::config::Config;
 
 const BLANK_LINE: [u8; 2] = [b'\n', b'\n'];
 const NOTE: &[u8; 109] = b"// This file was generated by [ts-rs](https://github.com/Aleph-Alpha/ts-rs). Do not edit this file manually.\n";
 
+struct Cleanup<'a>(&'a Config);
+impl<'a> Drop for Cleanup<'a> {
+    fn drop(&mut self) {
+        _ = fs::remove_file(self.0.output_directory.join(FILE_NAME));
+    }
+}
+
```

Comment:
Is there a specific reason for `Cleanup` to be its own struct rather than having `impl<'a> Drop for Config<'a>` as it was with `Args`?

Reply:
I think there was, though I can't completely recall.  
Maybe I moved it from `Args` into `Cleanup` to be able to destructure `Args`, though we may be able to move that back into `Config` if you prefer that.

Definetely like the idea of using a `Drop` impl like that, nice idea!

Reply:
> I think there was, though I can't completely recall.

Just found it, turns out that ´Deserialize´ moves data out of `Config` which isn't allowed if `Config` implements `Drop`

Reply:
I figured out a way past it, I changed the code to go back to using `Args` and changed `Config` so that it'll only be used to read the config file

Reply:
On a completely unrelated note, I never realized just how slow my home computer is until I ran `cargo test` and the `recursion_limit` tests took over 3 minutes to run xD